packages/flutter/lib/src/material/menu_style.dart

```diff
@@ -0,0 +1,447 @@
+/// can resolve to different colors depending on if the menu is pressed,
+/// hovered, focused, disabled, etc.
+///
+/// These properties can override the default value for just one state or all of
+/// them. For example to create a [MenuButton] whose background color is the
+/// color scheme’s primary color with 50% opacity, but only when the menu is
+/// pressed, one could write:
+///
+/// ```dart
+/// MenuButton(
+///   menuStyle: MenuStyle(
+///     backgroundColor: MaterialStateProperty.resolveWith<Color?>(
+///       (Set<MaterialState> states) {
+///         if (states.contains(MaterialState.focused)) {
+///           return Theme.of(context).colorScheme.primary.withOpacity(0.5);
+///         }
+///         return null; // Use the component's default.
+///       },
+///     ),
+///   ),
+///   menuChildren: const <Widget>[ /* ... */ ],
+///   child: const Text('Fly me to the moon'),
+/// ),
+/// ```
+///
+/// In this case the background color for all other menu states would fallback
+/// to the [MenuButton]'s default values. To unconditionally set the menu's
+/// [backgroundColor] for all states one could write:
+///
+/// ```dart
+/// const MenuButton(
+///   menuStyle: MenuStyle(
+///     backgroundColor: MaterialStatePropertyAll<Color>(Colors.green),
+///   ),
+///   menuChildren: <Widget>[ /* ... */ ],
+///   child: Text('Let me play among the stars'),
+/// ),
+/// ```
+///
+/// To configure all of the application's menus in the same
+/// way, specify the overall theme's `menuTheme`:
+///
+/// ```dart
+/// MaterialApp(
+///   theme: ThemeData(
+///     menuTheme: const MenuThemeData(
+///       style: MenuStyle(backgroundColor: MaterialStatePropertyAll<Color>(Colors.red)),
+///     ),
+///   ),
+///   home: const MyAppHome(),
+/// ),
+/// ```
+///
+/// See also:
+///
+///  * [MenuButtonTheme], the theme for [MenuButton]s and [MenuItemButton]s.
+///  * [ButtonStyle], a similar configuration object for button styles.
+@immutable
+class MenuStyle with Diagnosticable {
+  /// Create a [MenuStyle].
+  const MenuStyle({
+    this.backgroundColor,
+    this.shadowColor,
+    this.surfaceTintColor,
+    this.elevation,
+    this.padding,
+    this.minimumSize,
+    this.fixedSize,
+    this.maximumSize,
+    this.side,
+    this.shape,
+    this.mouseCursor,
+    this.visualDensity,
+    this.alignment,
+  });
+
+  /// The menu's background fill color.
+  final MaterialStateProperty<Color?>? backgroundColor;
+
+  /// The shadow color of the menu's [Material].
+  ///
+  /// The material's elevation shadow can be difficult to see for
+  /// dark themes, so by default the menu classes add a
+  /// semi-transparent overlay to indicate elevation. See
+  /// [ThemeData.applyElevationOverlayColor].
+  final MaterialStateProperty<Color?>? shadowColor;
+
+  /// The surface tint color of the menu's [Material].
+  ///
+  /// See [Material.surfaceTintColor] for more details.
+  final MaterialStateProperty<Color?>? surfaceTintColor;
```

Comment:
CC @darrenaustin 
Presumably the API doc for Material.surfaceTintColor will explain that using Colors.transparent means "no surface tint".

Reply:
Yup. I am prepping a PR with this support as we type.